### PR TITLE
🌐 🔠 📄 🐢 Translate docs to Emoji 🥳 🎉 💥 🤯 🤯

### PR DESCRIPTION
**Pull request by LeeeeT**
**I just really love 💕 🤩 the idea of adding some emojis 😄😁😅 here and there across the 📄 docs 📄 (like [in here](https://fastapi.tiangolo.com/async) for example). So I decided to write ✍ the entire documentation out of emojis 💥🎉🌟. All these words 🤮🤢 and... fancy-looking letters.. 👎👎👎 ugh... That's so boring 🥱. I believe reading documentation made of emojis will definitely (don't argue 😛🤪) be much funnier 😃 and easier 😻 for beginners. Just take a look at a new awesome  🔥 [async page](https://6426a9b35d5fc100b1db0463--fastapi.netlify.app/emj/async/) 🔥. So here it is, don't thank me. 💥💯💫⭐**

### CODE REVIEW

**Review by timoniq — APPROVED** [review]

**Review by axd1x8a — APPROVED** [review]

File: docs/emj/docs/features.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# 📇 🏠
+
+## 📇 🏠 FastAPI
+
+**FastAPI** 💁‍♂️ 👆 📄:
+
+### ⚙️ 🈺 🐩
+
+* <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank"><strong>OpenAPI</strong></a> 🏗 API, ✅ 🙂📛 <abbr title="💭   HTTP-👩‍🔬 ✅ : POST, GET, PUT, DELETE">🛠</abbr> <abbr title="💭: 🔗, 🛣, '🍵', ♒️.">🌌</abbr>, 🔢, ✔ 💪, 💂‍♂, ♒️.
+
```
Comment (MadL1me) [inline docs/emj/docs/features.md:10]: I think "Security" in native emoji is more like "👮" then your translated "💂‍♂". However, the other parts is perfect, I like it

**Review by MadL1me — CHANGES_REQUESTED** [review]

**Review by MadL1me — APPROVED** [review]

**Review by yallxe — APPROVED** [review]

**Review by axd1x8a — CHANGES_REQUESTED** [review]
It looks👀 like the markup📝 is a bit🤏 broken💔, making it hard😅 to read📖 such great👍 documentation📝.
https://6320bdb2dc53d17efc1ba804--fastapi.netlify.app/emj/
image
image

**Review by Dragonprod — APPROVED** [review]

**Review by luwqz1 — APPROVED** [review]

**Review by kiriharu — APPROVED** [review]

**Review by alexcher-im — APPROVED** [review]

**Review by kesha1225 — APPROVED** [review]

**Review by dhvcc — APPROVED** [review]

**Review by KurimuzonAkuma — APPROVED** [review]

File: docs/emj/docs/tutorial/background-tasks.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# 🖥 🎯
```
Comment (vypivshiy) [inline docs/emj/docs/tutorial/background-tasks.md:1]: image

**Review by vypivshiy — APPROVED** [review]

**Review by txlyre — APPROVED** [review]

**Review by rmdlv — APPROVED** [review]

File: docs/emj/mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    link: https://www.linkedin.com/in/tiangolo
+  - icon: fontawesome/brands/dev
+    link: https://dev.to/tiangolo
+  - icon: fontawesome/brands/medium
+    link: https://medium.com/@tiangolo
+  - icon: fontawesome/solid/globe
+    link: https://tiangolo.com
+  alternate:
+  - link: /
+    name: en - English
```
Comment (CringeDeveloper) [inline docs/emj/mkdocs.yml:110]: Maybe you should use country flag instead?

**Review by ichetiva — APPROVED** [review]

**Comment by tiangolo** [comment]
Hey there! Is this coming from a joke or something somewhere else? :sweat_smile: 

I imagine that given all the approvals and interactions, but I didn't see the original joke... :shrug:

**Comment by FeeeeK** [comment]
> @FeeeeK 🔥🔥🔥 Fixed 🔥🔥🔥

Great💫, I think this is a very nice👍 change, making the documentation📝 much friendlier🙌. It's a pity that the previous⏮️ version of the `Concurrency and async / await` documentation was removed❌, I really miss it🥲.

**Review by axd1x8a — APPROVED** [review]

**Comment by mrMiiao** [comment]
Really 😃😃😃 cool 😍😍🙂 idea 👍🧠🫦 bro 🥵😎🥵!!!

**Review by KurimuzonAkuma — APPROVED** [review]

**Comment by FeeeeK** [comment]
@tiangolo You see right through us. Forgive us this joke. We're just really sad about the decision to remove the emojis in #3273. They made the documentation unusual and stand out from the documentation of other frameworks. That's why we wanted to bring back that distinctive feature, so we'd be very happy if you could merge this pull request.❤️

**Comment by LeeeeT** [comment]
I think I could do a full translation of all the pages as well as fix a few formatting issues if you would ever consider adding an emoji version of the documentation, just for fun 😊

**Review by d1snin — APPROVED** [review]

**Review by vypivshiy — APPROVED** [review]

**Review by eoftgge — APPROVED** [review]

**Review by baseplate-admin — APPROVED** [review]

**Review by adomerlee — APPROVED** [review]

**Review by VodoGamer — APPROVED** [review]

**Review by notdanhan — APPROVED** [review]
I hate it
LGTM

**Comment by tiangolo** [comment]
So... I'm considering adding this for April Fool's Day. That's next April 1st. 🤓😂

**Comment by LeeeeT** [comment]
Cool! 🥳 I'm going to finish up translating the remaining pages and prepare this for merging as promised. 😄

**Comment by BrycensRanch** [comment]
LGTM

**Comment by FeeeeK** [comment]
I tried to send a review, but for some reason github just gives an error, I have already contacted support, I hope they will help, but for now please wait a bit.

**Review by dikiaap — APPROVED** [review]

**Review by notdanhan — APPROVED** [review]
LGTM :+1:

**Review by baseplate-admin — APPROVED** [review]
LGTM

**Comment by FeeeeK** [comment]
Now I know that github doesn't like this

**Comment by LeeeeT** [comment]
@FeeeeK It's probably for the best... 😐

**Review by tiangolo — APPROVED** [review]
I'm actually impressed. You did an amazing job here. The consistency of the translation from the original is great. 🤓😎

---

I moved the content from `emj` to`em`, that simplifies the URLs and makes using the internal scripts simpler (to run the docs live locally, etc).

`em` is not assigned to any other language so that's fine. https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes#Table_of_all_possible_two_letter_codes

---

I imagine you automatized parts of it, I would love to know how you did it!

---

I'm also cracking up. 😂

---

Thank you for your contribution! 🙇🍰

**Comment by FeeeeK** [comment]
> ---
> 
> I imagine you automatized parts of it, I would love to know how you did it!
> 
> ---
@tiangolo For all this, regexes were used to separate the text from code blocks, and then the translation through Yandex translator ([link](https://translate.yandex.com/?source_lang=en&target_lang=emj)), and after all this, I fixed some inaccuracies and broken emojis. I think this can also be automated and converted to GitHub action, but we need to make it a little more stable.

**Comment by LeeeeT** [comment]
Yea, it required some reverse engineering and use of private API in order to automate translating with Yandex. 😅 And my script for extracting readable and translatable text from `md` files is... well, it's better not to see it. 😹😹 Also, Yandex translator is far from ideal and some manual editing is required. So I don't think it should (and can) be automated. 🫤

Anyway, I want to thank all the participants of this pull request. ❤️ You guys made the prank even funnier. 🤪🥰

**Pull request merged**